src/error.rs

```diff
@@ -1,142 +1,36 @@
+use std::panic::Location;
```

Comment:
What's the point of keeping this module then in the end? The `OsError` sounds like it should be in the platform module, since it depends on the platform. We could maybe name it differently, like `PlatformError`.